sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp

```diff
@@ -23,210 +23,238 @@
 #include <string>
 #include <vector>
 
-namespace Azure { namespace Security { namespace KeyVault {
+namespace Azure {
+  namespace Security {
+    namespace KeyVault {
 
+      namespace Keys {
+        namespace Cryptography {
   namespace _detail {
-    class KeyVaultProtocolClient;
+    /**
+     * @brief The internal access to the cryptography client.
+     *
+     */
+    class CryptoClientInternalAccess;
   } // namespace _detail
 
-  namespace Keys { namespace Cryptography {
+  /**
+   * @brief A client used to perform cryptographic operations with Azure Key Vault keys.
+   *
+   */
+  class CryptographyClient final {
+  protected:
+    Azure::Core::Url m_keyId;
+    std::string m_apiVersion;
+    std::shared_ptr<Azure::Core::Http::_internal::HttpPipeline> m_pipeline;
+
+  private:
+    // Provide private-access to the internal layer
+    friend class Azure::Security::KeyVault::Keys::Cryptography::_detail::CryptoClientInternalAccess;
+
```

Comment:
Oh... the whole purpose of `CreateCryptographyClient` is to use it as a friend class?

Reply:
I am having trouble understanding the intention behind this, probably because I am not as familiar with the KeyVault code base. Can you help me understand what requires us to do this indirection of access, and why KeyClient can't construct a `CryptographyClient` directly?

Is it because the pipeline parameter is internal, and hence the ctor has to be private?